### PR TITLE
WIP: Add OTEL receiver code and example OTEL collector code

### PR DESCRIPTION
**Pull request by rockdaboot**
This PR is based on #84 and is a PoC for ongoing work (cc @dmathieu).

### CODE REVIEW

File: otel/README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ Technically, the receiver and collector code can both live in separate repositor
 
 To get the `ocb` tool (OTEL collector builder, amend version):
 ```
-curl --proto '=https' --tlsv1.2 -fL -o ocb https://github.com/open-telemetry/opentelemetry-collector/releases/download/cmd%2Fbuilder%2Fv0.106.0/ocb_0.106.0_linux_amd64
+curl --proto '=https' --tlsv1.2 -fL -o ocb https://github.com/open-telemetry/opentelemetry-collector/releases/download/cmd%2Fbuilder%2Fv0.107.0/ocb_0.107.0_linux_amd64
```
Comment (dmathieu) [inline otel/README.md:8]: It looks like the builder is still on 0.106, there's no 0.107 yet.
https://github.com/open-telemetry/opentelemetry-collector/releases/tag/cmd%2Fbuilder%2Fv0.106.1

File: otel/builder-config.yaml [hunk 2]
```diff
@@ -2,21 +2,21 @@ dist:
   name: profiling-collector
   description: Profiling OTel Collector distribution
   output_path: ./collector
-  otelcol_version: 0.106.0
+  otelcol_version: 0.107.0
```
Comment (dmathieu) [inline otel/builder-config.yaml:5]: We should use 0.109.0, since this is where the services are able to handle profiles.

Comment (rockdaboot) [reply otel/builder-config.yaml]: Updated to 109

**Comment by rockdaboot** [comment]
Cleaned up version of this PR is #160. Closing in favor of it.

**Pull request closed without merging**